File: leetcodepractice/1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let newWord = ""; // newWord = 'apbqrs'

    for (let i = 0; i < Math.max(word1.length, word2.length); i++){ // i = 2, max 4
        if (i < word1.length) newWord += word1[i]; // 2 < 2 ? 
        if (i < word2.length) newWord += word2[i]; // 2 < 4 ?
    }

    return newWord;
};

File: leetcodepractice/1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        new_word = ""
        word1_length = len(word1)
        word2_length = len(word2)

        if word1_length == word2_length:
            for i in range(max(len(word1), len(word2))):
                new_word += word1[i]
                new_word += word2[i]
        else:
            min_index = min(word1_length, word2_length)
            for i in range(min_index):
                new_word += word1[i]
                new_word += word2[i]
            if word1_length < word2_length:
                new_word += word2[word1_length:word2_length]
            else:
                new_word += word1[word2_length:word1_length]
        return new_word
